server/src/main/scala/org/http4s/server/middleware/ChunkAggregator.scala

```diff
@@ -2,19 +2,32 @@ package org.http4s
 package server
 package middleware
 
+import cats.data.NonEmptyList
 import fs2._
 import fs2.interop.cats._
 import org.http4s.headers._
 
 object ChunkAggregator {
-  def apply(service: HttpService): HttpService = service.flatMapF[MaybeResponse] {
-    case response: Response =>
-      response.body.runLog.map { bytes =>
-        if (bytes.nonEmpty) {
-          val h = response.headers.put(`Content-Length`(bytes.length.toLong))
-          response.copy(body = Stream.emits(bytes), headers = h)
-        } else response
+  def apply(service: HttpService): HttpService =
+    service.flatMapF[MaybeResponse] {
+      case Pass => Task.now(Pass)
+      case response: Response =>
+        response.body.runLog.flatMap { fullBody =>
```

Comment:
Might be able to make this more efficient by foldMapping into a `ByteVector`, and creating a `ByteVectorChunk` from that.  That would avoid some copying of bytes.  But I think this is correct.